CRM/Core/Session.php

```diff
@@ -171,9 +163,9 @@ public function reset($all = 1) {
       unset($this->_session[$this->_key]);
     }
     else {
-      $this->_session = [];
+      $this->_session[$this->_key] = [];
+      unset($this->_session);
```

Comment:
won't this `unset` nuke the `$_SESSION`, because `$this->_session` is a reference to that variable?

Reply:
Yes.

Short Example: 
$a = 1;
$b = &$a;
$b++;

echo $a; // display 2
echo $b; // display 2

unset($b);
echo $a; // still display 2
echo $b; // undefined

Reply:
oh - right, make sense, merci :)